src/main/scala/performanceanalysis/DateParser.scala

```diff
@@ -0,0 +1,55 @@
+package performanceanalysis
+
+import java.time.{LocalDate, LocalDateTime, LocalTime}
+
+import performanceanalysis.DateParser._
+
+import scala.util.matching.Regex.Match
+import scala.util.{Failure, Success, Try}
+
+object DateParser {
+
+  private val regex = """(\d{1,2}|\d{4})[ -/\.](\d{1,2})[ -/\.](\d{1,2}|\d{4})[ T]( \d|\d{2})[:.](\d{2})[.:](\d{2})(\.\d{0,9})?""".r
+
+  private def parser(iYear: Int, iMonth: Int, iDay: Int): DateParser = new DateParser(iYear, iMonth, iDay)
+
+  val parseYMD: (String) => Option[LocalDateTime] = parser(1, 2, 3).parser
+
+  val parseDMY: (String) => Option[LocalDateTime] = parser(3, 2, 1).parser
+
+  val parseMDY: (String) => Option[LocalDateTime] = parser(3, 1, 2).parser
+
+}
+
+private class DateParser(iYear: Int, iMonth: Int, iDay: Int) {
+
+  def rawDateParser(m: Match): LocalDate = {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [You should not name methods after their defining object](https://www.codacy.com/app/scala-academy/performance-analysis/file/2612051301/issues/source?bid=3220915&fileBranchId=3304352#l26)